### DISCUSSION TITLE
How to add WatchOS to Fastlane?

### DISCUSSION THREAD

**Discussion by ir-fuel** [fastlane/fastlane#19272] [Q&A]
I have a correctly functioning Fastlane setup for my iOS app. Now I added a WatchOS extension and things have started breaking.

First error I got is:

```
❌  error: No profiles for 'com.***.***.watchkitapp.watchkitextension' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.***.***.watchkitapp.watchkitextension'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'WatchOS Extension' from project '****')
```

So I changed to manual signing in my WatchOS app and extension. But then I need provisioning profiles. Tried running `nuke` but it only regenerated my iOS profiles.

Checked the docs and I found out I can specify multiple identifiers in `match`, so I added the following lane:

```
  lane :certificates do
    match(app_identifier: ["com***.***", "com***.***.watchkitapp", "com***.***.watchkitapp.watchkitextension"])
  end  
```

However this doesn't work. I get the following error:
```
 An app with that bundle ID needs to exist in order to create a provisioning profile for it
```

Which is logical because the watchkitapp and extension don't exist as standalone bundle ID's in my developer account.

So how do I fix this?

Thanks

**Comment by lucgrabowski** [comment]
Hi @ir-fuel, 
you need to create App Ids to be able to create provisioning profiles. 
You could go to Developer Center - Certificates, Identifiers and Profiles, register a new Identifier - App Id
Or you could do this with `produce`, in that case you should check the meaning of `skip_itc` flag https://docs.fastlane.tools/actions/produce/ - You probably don't want to create an app on AppStoreConnect for extension

**Reply by ir-fuel** [reply]
So you need to create app ids on the Developer Portal for WatchOS extensions? (this is not a standalone app). Normally this is not required by Apple if I am not mistaken?

**Reply by lucgrabowski** [reply]
Yes, you need app ids for extensions, it's required by Apple.
